client/payment-methods/style.scss

```diff
@@ -1,5 +1,19 @@
 .payment-methods {
-	position: relative;
+ 	position: relative;
+
+	&__header {
+		justify-content: space-between;
+
+		.components-dropdown-menu__toggle.has-icon {
+			padding: 0;
+			min-width: unset;
+		}
+	}
+
+	&__header__heading {
```

Comment:
I know we're not exactly using BEM, but [in BEM](https://en.bem.info/methodology/quick-start/#nesting-1):

> An element is always part of a block, not another element. This means that element names can't define a hierarchy such as block__elem1__elem2.

```suggestion
	&__heading {
```

And in [Calypso CSS/Sass Coding Guidelines](https://github.com/Automattic/wp-calypso/blob/trunk/docs/coding-guidelines/css.md#classes):

> We keep one level of prefix for single components present in the class name: .site__title instead of .site .title. This may seem verbose but it provides more flexibility, clear direct selectors, and immediate recognizability of its role in a large codebase like Calypso. Classes are not just for style purposes, they should provide meaning to the reader parsing the document.

Reply:
I initially committed this - I just reverted. When tested, I noticed that the styles are not applied correctly.

With `&__heading`
![Screen Shot 2021-07-20 at 6 32 49 AM](https://user-images.githubusercontent.com/273592/126317087-b6340305-b5fd-4ff2-9fdb-951aa9ffc3f8.png)

With `&__header__heading`
![Screen Shot 2021-07-20 at 6 32 20 AM](https://user-images.githubusercontent.com/273592/126317101-05507d6b-ee54-4de8-9a2c-f2103da7802c.png)